src/gui/issueswidget.cpp

```diff
@@ -209,9 +226,47 @@ void IssuesWidget::slotOpenFile(QTreeWidgetItem *item, int)
 
 void IssuesWidget::slotProgressInfo(const QString &folder, const ProgressInfo &progress)
 {
-    if (progress.status() == ProgressInfo::Starting) {
-        // The sync is restarting, clean the old items
-        cleanItems(folder);
+    if (progress.status() == ProgressInfo::Reconcile) {
+        // Wipe all non-persistent entries - as well as the persistent ones
+        // in cases where a local discovery was done.
+        auto f = FolderMan::instance()->folder(folder);
+        if (!f)
+            return;
+        const auto &engine = f->syncEngine();
+        const auto style = engine.lastLocalDiscoveryStyle();
+        cleanItems([&](QTreeWidgetItem *item) {
+            if (ProtocolItem::extraData(item).folderName != folder)
+                return false;
+            if (style == LocalDiscoveryStyle::FilesystemOnly)
+                return true;
+            if (!persistsUntilLocalDiscovery(item))
+                return true;
+
+            // Definitely wipe the entry if the file no longer exists
+            if (!QFileInfo(f->path() + ProtocolItem::extraData(item).path).exists())
```

Comment:
I still think this might be wrong.

Reply:
Can you elaborate again? Now that this covers only `FileIgnored` and `Conflict` statuses that are local this seemed like a correct safeguard to me.

Reply:
So won't happen for errors  (eg, download error).
And file ignored on the server are already not in there?

Then i guess it is ok.

Reply:
Yes, exactly. This only applies to items that can stay around until a folder is locally rediscovered: currently only local ignores and conflicts.